packages/crayons-core/src/components/tag/tag.tsx

```diff
@@ -76,32 +96,77 @@ export class Tag {
     if (this.disabled || !this.closable) {
       return;
     }
-    const { value, text } = this;
-    this.fwClosed.emit({ value, text });
+    const { value, text, index } = this;
+    this.fwClosed.emit({ value, text, index });
     e.stopPropagation();
   };
 
+  private renderLabel() {
+    return (
+      <div
+        class='ellipsis'
+        ref={(el) => (this.divLabel = el)}
+      >
+        <span
+          class={{
+            primary: !!this.secondaryText,
+            content: true,
+            'end-padding': !this.secondaryText && !this.closable,
+          }}>{this.text}</span>
+        {this.secondaryText && <span class={`secondary content ${!this.closable ? 'end-padding' : ''}`}>{this.secondaryText}</span>}
+      </div>
+    );
+  }
+
   renderContent() {
     switch (this.variant) {
       case 'standard':
         return <span class='content'>{this.text}</span>;
       case 'avatar': {
         return [
-          <fw-avatar size='xsmall' {...this.graphicsProps}></fw-avatar>,
-          <span class='content'>{this.text}</span>,
+          <fw-avatar mode={this.state === 'error' ? this.state : 'dark'} size='xsmall' {...this.graphicsProps}></fw-avatar>,
```

Comment:
The applications that are using exsiting tag component with avatar variant will see their layout change with this change ?. can we ensure the existing apps don't break or can we call this out

Reply:
@arvindanta These changes will add an ellipsis if the text overflows over the available width and shows the tooltip only when ellipsis occur. Existing tag component allows text to be overflown over the available width which is not desired. Can we consider having this change in our tag component by default or should we enable this via new prop?

Reply:
@rihansiddhi can u pls check wit the designers and see if for regular tag usecases, should we have tooltip for texts that overflow